client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Transports.ts

```diff
@@ -1,28 +1,41 @@
 import { DataReceived, TransportClosed } from "./Common"
 import { IHttpClient } from "./HttpClient"
+import { HttpError } from "./HttpError"
 
 export enum TransportType {
     WebSockets,
     ServerSentEvents,
     LongPolling
 }
 
+export const enum TransferMode {
+    Text = 1,
+    Binary
+}
+
 export interface ITransport {
-    connect(url: string): Promise<void>;
+    connect(url: string, requestedTransferMode: TransferMode): Promise<void>;
     send(data: any): Promise<void>;
     stop(): void;
     onDataReceived: DataReceived;
     onClosed: TransportClosed;
+    transferMode(): TransferMode;
```

Comment:
I am thinking about removing this and just returning Promise<TransferMode> from connect. Same for connection.